models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/api/MiniMaxApi.java

```diff
@@ -0,0 +1,890 @@
+			this(messages, model, null,  null, null, null,
+					null, null, null, false, temperature, null,
+					null, null);
+		}
+
+		/**
+		 * Shortcut constructor for a chat completion request with the given messages, model and control for streaming.
+		 *
+		 * @param messages A list of messages comprising the conversation so far.
+		 * @param model ID of the model to use.
+		 * @param temperature What sampling temperature to use, between 0 and 1.
+		 * @param stream If set, partial message deltas will be sent.Tokens will be sent as data-only server-sent events
+		 * as they become available, with the stream terminated by a data: [DONE] message.
+		 */
+		public ChatCompletionRequest(List<ChatCompletionMessage> messages, String model, Float temperature, boolean stream) {
+			this(messages, model, null,  null, null, null,
+					null, null, null, stream, temperature, null,
+					null, null);
+		}
+
+		/**
+		 * Shortcut constructor for a chat completion request with the given messages, model, tools and tool choice.
+		 * Streaming is set to false, temperature to 0.8 and all other parameters are null.
+		 *
+		 * @param messages A list of messages comprising the conversation so far.
+		 * @param model ID of the model to use.
+		 * @param tools A list of tools the model may call. Currently, only functions are supported as a tool.
+		 * @param toolChoice Controls which (if any) function is called by the model.
+		 */
+		public ChatCompletionRequest(List<ChatCompletionMessage> messages, String model,
+				List<FunctionTool> tools, Object toolChoice) {
+			this(messages, model, null, null, null, null,
+					null, null, null, false, 0.8f, null,
+					tools, toolChoice);
+		}
+
+				/**
+		 * Shortcut constructor for a chat completion request with the given messages, model, tools and tool choice.
+		 * Streaming is set to false, temperature to 0.8 and all other parameters are null.
+		 *
+		 * @param messages A list of messages comprising the conversation so far.
+		 * @param stream If set, partial message deltas will be sent.Tokens will be sent as data-only server-sent events
+		 * as they become available, with the stream terminated by a data: [DONE] message.
+		 */
+		public ChatCompletionRequest(List<ChatCompletionMessage> messages, Boolean stream) {
+			this(messages, null, null,  null, null, null,
+					null, null, null, stream, null, null,
+					null, null);
+		}
+
+		/**
+		 * Helper factory that creates a tool_choice of type 'none', 'auto' or selected function by name.
+		 */
+		public static class ToolChoiceBuilder {
+			/**
+			 * Model can pick between generating a message or calling a function.
+			 */
+			public static final String AUTO = "auto";
+			/**
+			 * Model will not call a function and instead generates a message
+			 */
+			public static final String NONE = "none";
+
+			/**
+			 * Specifying a particular function forces the model to call that function.
+			 */
+			public static Object FUNCTION(String functionName) {
```

Comment:
why is the method name in all capital letters?

Reply:
resolved in PR #744